private/byte.rkt

```diff
@@ -109,3 +116,279 @@
           (check-equal? (byte-ref abcdefgh 5) f)
           (check-equal? (byte-ref abcdefgh 6) g)
           (check-equal? (byte-ref abcdefgh 7) h))))))
+
+(define (byte-and left right)
+  (apply byte
+         (for/list ([i (in-range 8)])
+           (bitwise-and (byte-ref left i)
+                        (byte-ref right i))))) 
+
+(module+ test
+  (test-case
+    "byte-and"
+    (for ([x (in-range 256)])
+      (check-equal? (byte-ref (byte-and x (byte 0 1 1 1 1 1 1 1)) 0) 0)
+      (check-equal? (byte-ref (byte-and (byte 0 1 1 1 1 1 1 1) x) 0) 0)
```

Comment:
Instead of having each check twice here, we could have a separate loop that checks the operation is always commutative.

```
(for* ([x (in-range 256)] [y (in-range 256)])
  (check-equal? (byte-and x y) (byte-and y x)))
```

You'll need to use `for*` instead of `for`. The `for` macro steps over each loop variable together, whereas `for*` acts like a nested for. The above `for*` loop is equivalent to this:

```
(for ([x (in-range 256)])
  (for ([y (in-range 256)])
    (check-equal? (byte-and x y) (byte-and y x))))
```